src/Sentry.Unity/MainThreadData.cs

```diff
@@ -79,7 +86,9 @@ public static void CollectData()
     {
         var sentrySystemInfo = SentrySystemInfo ?? SentrySystemInfoAdapter.Instance;
 
-        MainThreadId = sentrySystemInfo.MainThreadId;
+        // Don't overwrite the MainThreadId if it's already been set
+        MainThreadId ??= sentrySystemInfo.MainThreadId;
```

Comment:
This operation is not atomic. How about using `Interlocked.CompareExchange`, with non-nullable mainThreadId that defaults to -1 in which case `IsMainThread` returns null, or even better, change the return value to an enum with "unknown" as one of the values if you care about that. If it doesn't matter, it can just return false until it's known.